bitnami/mysql/templates/secrets.yaml

```diff
@@ -1,13 +1,38 @@
+{{- if (not .Values.root.existingSecret) -}}
 apiVersion: v1
 kind: Secret
 metadata:
   name: {{ template "fullname" . }}
   labels:
-    app: {{ template "fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+    app: "{{ template "name" . }}"
+    chart: {{ template "mysql.chart" . }}
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
 type: Opaque
 data:
-  mysql-root-password: {{ default "" .Values.mysqlRootPassword | b64enc | quote }}
-  mysql-password: {{ default "" .Values.mysqlPassword | b64enc | quote }}
+  {{- if .Values.root.password }}
+  mysql-root-password: "{{ .Values.root.password | b64enc }}"
+  {{- else if (not .Values.root.forcePassword) }}
+  mysql-root-password: "{{ randAlphaNum 10 | b64enc }}"
```

Comment:
You can also use `randAscii` (see https://github.com/Masterminds/sprig/blob/master/docs/strings.md#randalphanum-randalpha-randnumeric-and-randascii)